base_layer/core/src/proof_of_work/monero_rx.rs

```diff
@@ -231,25 +231,343 @@ fn verify_root(monero_data: &MoneroData) -> Result<(), MergeMineError> {
     Ok(())
 }
 
-fn merged_mining_subfield(header: &BlockHeader) -> SubField {
-    let hash = header.merged_mining_hash();
-    let depth = 0;
-    SubField::MergeMining(VarInt(depth), Hash::from(from_slice(&hash)))
-}
-
 fn verify_header(header: &BlockHeader, monero_data: &MoneroData) -> Result<(), MergeMineError> {
-    if !(monero_data
-        .coinbase_tx
-        .prefix
-        .extra
-        .0
-        .contains(&merged_mining_subfield(header)))
-    {
+    let mut found = false;
+    for item in monero_data.coinbase_tx.prefix.extra.0.clone() {
+        match item {
+            SubField::MergeMining(x, y) => {
+                let compare = Hash::from(from_slice(&header.merged_mining_hash()));
+                if (x, y) == (VarInt(0), compare) {
```

Comment:
I don't hate this "trick" - but if you don't have to use a trick in the first place usually results in clearer code.

```suggestion
                if depth == VarInt(0) && merge_mining_hash == hash {
```

Reply:
The problem was that I needed to match a `typeof()` on a `enum` which is why I did it this way.